@here/harp-datasource-protocol/lib/Expr.ts

```diff
@@ -75,16 +118,56 @@ export abstract class Expr {
 
             case "get":
                 if (node[2] !== undefined) {
-                    return Expr.makeCallExpr(op, node);
+                    return Expr.makeCallExpr(op, node, referenceResolverState);
                 }
                 if (typeof node[1] !== "string") {
                     throw new Error(`expected the name of an attribute`);
                 }
                 return new VarExpr(node[1]);
 
+            case "ref":
```

Comment:
could you move this to its own function.

Reply:
done